lms/security.py

```diff
@@ -199,6 +208,49 @@ def permits(self, request, _context, permission):
         return _permits(self.identity(request), permission)
 
 
+class EmailSettingsSecurityPolicy:
+    def __init__(self, email_unsubscribe_service: EmailUnsubscribeService):
+        self.helper = AuthTktCookieHelper(
+            secret="secret",
+            cookie_name="email.settings",
+            secure=False,
+            timeout=60 * 5,
+            reissue_time=0,
+            max_age=60 * 5,
+            path="/email/settings",
+            http_only=True,
+            wild_domain=False,
+            parent_domain=False,
+            domain="localhost",
+        )
+        self.email_unsubscribe_service = email_unsubscribe_service
+
+    def identity(self, request):
+        if token := request.params.get("token"):
+            try:
+                return self.email_unsubscribe_service.decode_token(token)["h_userid"]
+            except InvalidJWTError:
+                return None
+
+        try:
+            return self.helper.identify(request)["userid"]
```

Comment:
Maybe this is already taken into consideration for production PRs, but other security policies seem to be returning a `dict` from the `identity` method, while this one returns a string. That makes some later checks fail.

I tried applying this change this for the case in which we get the token in the first place:

```diff
- return self.email_unsubscribe_service.decode_token(token)["h_userid"]
+ return self.email_unsubscribe_service.decode_token(token)
```

But `self.helper.identify(request)["userid"]` still returns a serialized/stringified version of the `dict` in that case.